MyString.java

```diff
@@ -49,8 +75,17 @@ public static boolean subsetOf(String str1, String str2) {
      * @return a string consisting of the characters of str, separated by spaces.
      */
     public static String spacedString(String str) {
-        //// Replace the following statement with your code
-        return null;
+        int length= str.length();
+        String newString="";
+        if (str.length()==0) {
+            return ""+"";
+        }
+        
+        for (int i=0; i<length-1; i++){
+            newString+="" +str.charAt(i)+ " ";
+        }
+        newString+=""+str.charAt(length-1);

```

Comment:
1.dont need ""  at start. 
2.better to put inside loop, see recitation 4, array print for details